src/AdoNetCore.AseClient/Internal/TypeMap.cs

```diff
@@ -37,27 +37,28 @@ internal static class TypeMap
             {DbType.Time, (value, length) => TdsDataType.TDS_BIGDATETIMEN}
         };
 
-        private static readonly Dictionary<Type, Func<object, int, TdsDataType>> NetTypeToTdsMap = new Dictionary<Type, Func<object, int, TdsDataType>>
+        private static readonly Dictionary<Type, AseDbType> NetTypeToAseDbTypeMap = new Dictionary<Type, AseDbType>
         {
-            {typeof(bool), (value, length) => TdsDataType.TDS_BIT},
-            {typeof(byte), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT1},
-            {typeof(sbyte), (value, length) => TdsDataType.TDS_INTN},
-            {typeof(short), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT2},
-            {typeof(ushort), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT2},
-            {typeof(int), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT4},
-            {typeof(uint), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT4},
-            {typeof(long), (value, length) => value == DBNull.Value ? TdsDataType.TDS_INTN : TdsDataType.TDS_INT8},
-            {typeof(ulong), (value, length) => value == DBNull.Value ? TdsDataType.TDS_UINTN : TdsDataType.TDS_UINT8},
-            {typeof(char), (value, length) => TdsDataType.TDS_CHAR},
-            {typeof(char[]), (value, length) => TdsDataType.TDS_CHAR},
-            {typeof(string), (value, length) => TdsDataType.TDS_LONGBINARY},
```

Comment:
This has become a VarChar - assume this is an intentional fix of a previous bug.

Reply:
It's an intentional fix but not for a previous bug. It's my assumption that by default, strings should be considered `varchar` rather than `univarchar` -- perhaps I should actually test this assumption however... 😜